os/arch/arm/src/armv7-m/up_blocktask.c

```diff
@@ -171,17 +171,14 @@ void up_block_task(struct tcb_s *tcb, tstate_t task_state)
 
 			/* Restore the MPU registers in case we are switching to an application task */
 #ifdef CONFIG_ARMV7M_MPU
-			/* Condition check : Update MPU registers only if this is not a kernel thread. */
-			if ((rtcb->flags & TCB_FLAG_TTYPE_MASK) != TCB_FLAG_TTYPE_KERNEL) {
 #if defined(CONFIG_APP_BINARY_SEPARATION)
-				for (int i = 0; i < 3 * MPU_NUM_REGIONS; i += 3) {
-					up_mpu_set_register(&rtcb->mpu_regs[i]);
-				}
+			for (int i = 0; i < 3 * MPU_NUM_REGIONS; i += 3) {
```

Comment:
This is only for user. Only stack protection is for both. Please check this.

Reply:
Done. Changes now only for stack protection.